### PR TITLE
Add types for node-hue-api npm package

### PR DESCRIPTION
**Pull request by fjmorel**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

### CODE REVIEW

**Comment by vvakame** [comment]
*node-hue-api/index.d.ts*

Checklist

* [X] is correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/node-hue-api - https://github.com/peter-murray/node-hue-api#readme
* [X] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (node-hue-api-tests.ts or node-hue-api-tests.tsx)
* [ ] pass the Travis CI test?

File: node-hue-api/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,1075 @@
+// Type definitions for node-hue-api 2.3.0
+// Project: https://github.com/peter-murray/node-hue-api
+// Definitions by: wind-rider <https://github.com/wind-rider/>, Frederic Morel <https://github.com/fjmorel>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+// Functions that are deprecated, not working, or returning callbacks instead of promises were not documented.
+
+declare module "node-hue-api" {
```
Comment (vvakame) [inline node-hue-api/index.d.ts]: could you remove this block?
https://github.com/Microsoft/types-publisher#fields-in-datadefinitionsjson
We recommend to use `ProperModule` in TypeScript 2.0.

**Comment by minestarks** [comment]
@fjmorel Thanks for the contribution. Do you plan to address the requested changes? Since this PR has been inactive for a long time I plan to close it out in a couple of days for housekeeping purposes if no further changes. Thank you.

**Comment by fjmorel** [comment]
Yes, I plan on addressing those. I've been swamped, but should have some time during Thanksgiving week.

**Comment by paulvanbrenk** [comment]
@fjmorel gentle ping

**Comment by fjmorel** [comment]
Deleted those lines (and the closing bracket).

**Comment by paulvanbrenk** [comment]
Thanks!

**Pull request merged**